sled-agent/types/src/instance.rs

```diff
@@ -47,6 +47,39 @@ pub struct InstanceEnsureBody {
     pub metadata: InstanceMetadata,
 }
 
+/// A request to attach a disk to an instance.
+#[derive(Clone, Debug, Serialize, Deserialize, JsonSchema)]
+pub struct InstanceDisk {
+    /// The disk's UUID.
+    pub disk_id: Uuid,
+    /// The logical slot number assigned to the disk in its database record.
+    pub slot: u8,
+    /// True if the disk is read-only.
+    pub read_only: bool,
+    /// A JSON representation of the Crucible volume construction request for
+    /// this attachment.
+    //
+    // This is marked as `NoDebug` because the VCR contains the volume's
+    // encryption keys.
+    pub vcr_json: NoDebug<String>,
```

Comment:
my assumption here is `NoDebug` means that it will be serialized/deserialized, just not printed to logs via the Debug derive, correct?

Reply:
Yup.